mat64/dense_arithmetic.go

```diff
@@ -641,9 +644,10 @@ func (m *Dense) Exp(a Matrix) {
 		scaling = 4
 	)
 
-	var small, power Dense
+	small := &Dense{}
 	small.Scale(math.Pow(2, -scaling), a)
-	power.Clone(&small)
+	power := &Dense{}
```

Comment:
My preference is concrete over pointer for internal things unless it stos you doing what you need to,do. That's what happened here.